### PR TITLE
CLOUD-486 Allow to adjust securityContext

### PR DESCRIPTION
**Pull request by blez**

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/percona-server-mongodb-operator?pullRequest=294) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-server-mongodb-operator?pullRequest=294) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-server-mongodb-operator?pullRequest=294) it.</sub>

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| init-deploy | passed |
| storage | passed |
| demand-backup | failed |
| monitoring | failed |
| limits | passed |
| scaling | failed |
| operator-self-healing | passed |
| one-pod | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| monitoring | passed |
| self-healing | passed |
| init-deploy | passed |
| operator-self-healing | passed |
| limits | passed |
| one-pod | failed |
| scaling | passed |
| demand-backup | failed |
| monitoring-2-0 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| operator-self-healing | passed |
| one-pod | failed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| storage | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| demand-backup | failed |
| arbiter | passed |
| service-per-pod | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| self-healing | passed |
| operator-self-healing | passed |
| one-pod | failed |
| demand-backup | failed |
| storage | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |

File: pkg/psmdb/backup/job.go [hunk 1]
```diff
@@ -19,23 +19,15 @@ func init() {
...
-func BackupCronJob(backup *api.BackupTaskSpec, crName, namespace, image, serviceAccountName string, imagePullSecrets []corev1.LocalObjectReference, sv *version.ServerVersion) *batchv1b.CronJob {
-	var fsgroup *int64
-	if sv.Platform == api.PlatformKubernetes {
-		var tp int64 = 1001
-		fsgroup = &tp
-	}
-
-	trueVar := true
-
+func BackupCronJob(backup *api.BackupTaskSpec, crName, namespace string, backupSpec api.BackupSpec, imagePullSecrets []corev1.LocalObjectReference, sv *version.ServerVersion) *batchv1b.CronJob {
```
Comment (delgod) [inline pkg/psmdb/backup/job.go]:
```suggestion
func BackupCronJob(backup *api.BackupTaskSpec, crName, namespace string, backupSpec api.BackupSpec, imagePullSecrets []corev1.LocalObjectReference) *batchv1b.CronJob {
```

File: pkg/psmdb/backup/agent.go [hunk 2]
```diff
@@ -17,13 +17,6 @@ import (
 )
 
 func AgentContainer(cr *api.PerconaServerMongoDB, sv *version.ServerVersion) corev1.Container {
```
Comment (delgod) [inline pkg/psmdb/backup/agent.go]:
```suggestion
func AgentContainer(cr *api.PerconaServerMongoDB) corev1.Container {
```

File: pkg/psmdb/statefulset.go [hunk 3]
```diff
@@ -30,12 +30,6 @@ var secretFileMode int32 = 288
 // StatefulSpec returns spec for stateful set
 // TODO: Unify Arbiter and Node. Shoudn't be 100500 parameters
 func StatefulSpec(m *api.PerconaServerMongoDB, replset *api.ReplsetSpec, containerName string, ls map[string]string, multiAZ api.MultiAZ, size int32, ikeyName string, sv *version.ServerVersion) (appsv1.StatefulSetSpec, error) {
```
Comment (delgod) [inline pkg/psmdb/statefulset.go]:
```suggestion
func StatefulSpec(m *api.PerconaServerMongoDB, replset *api.ReplsetSpec, containerName string, ls map[string]string, multiAZ api.MultiAZ, size int32, ikeyName string) (appsv1.StatefulSetSpec, error) {
```

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | failed |
| storage | failed |
| init-deploy | failed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | failed |
| storage | failed |
| init-deploy | failed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | failed |
| storage | failed |
| init-deploy | failed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| self-healing | passed |
| demand-backup | failed |
| monitoring | passed |
| limits | passed |
| monitoring-2-0 | failed |
| scaling | passed |
| operator-self-healing | passed |
| one-pod | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| operator-self-healing | passed |
| one-pod | failed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| storage | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| demand-backup | failed |
| arbiter | passed |
| service-per-pod | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| demand-backup | failed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| scaling | passed |
| monitoring-2-0 | failed |
| operator-self-healing | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| operator-self-healing | failed |
| storage | passed |
| monitoring | passed |
| monitoring-2-0 | failed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | failed |
| init-deploy | passed |
| storage | failed |
| demand-backup | failed |
| monitoring | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| limits | passed |
| monitoring-2-0 | passed |
| upgrade | failed |
| scaling | passed |
| one-pod | failed |
| arbiter | passed |
| service-per-pod | passed |
| security-context | failed |
| upgrade-consistency | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| storage | passed |
| demand-backup | failed |
| monitoring | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| security-context | failed |
| scheduled-backup | failed |
| upgrade | failed |
| operator-self-healing | passed |
| one-pod | failed |
| upgrade-consistency | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| security-context | failed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| self-healing | passed |
| operator-self-healing | passed |
| one-pod | failed |
| demand-backup | failed |
| scheduled-backup | failed |
| upgrade | failed |
| upgrade-consistency | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| init-deploy | passed |
| monitoring | passed |
| operator-self-healing | passed |
| limits | passed |
| monitoring-2-0 | passed |
| one-pod | passed |
| scaling | passed |
| arbiter | passed |
| service-per-pod | passed |
| security-context | failed |
| demand-backup | failed |
| scheduled-backup | failed |
| upgrade | failed |
| upgrade-consistency | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| init-deploy | passed |
| limits | passed |
| operator-self-healing | passed |
| monitoring | passed |
| scaling | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| security-context | failed |
| demand-backup | passed |
| scheduled-backup | passed |
| upgrade | passed |
| upgrade-consistency | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| init-deploy | passed |
| storage | passed |
| demand-backup | passed |
| monitoring | passed |
| limits | passed |
| scheduled-backup | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| upgrade | failed |
| arbiter | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| service-per-pod | passed |
| security-context | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-294-1c7dbceb

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| self-healing | passed |
| operator-self-healing | passed |
| one-pod | passed |
| storage | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| security-context | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| upgrade | passed |
| upgrade-consistency | passed |

**Pull request merged**